dist/s_clang_scan

```diff
@@ -36,25 +38,35 @@ $scan $args ninja wt > ../$t_out 2>&1
 cd ..
 
 # Strip comments from the diff file.
-cat dist/s_clang_scan.diff | sed '/^#/d' > $t_diff_filtered
+cat dist/s_clang_scan.diff | sed '/^#/d' | LC_ALL=C sort > $t_diff_filtered
 
 # Compare the set of errors/warnings and exit success when they're expected.
 # clang_scan outputs errors/warnings with a leading file path followed by a
 # colon, followed by the string "error:" or "warning:".
 cat $t_out |
     grep -E '^[a-z\./].*error: |^[a-z\./].*warning: ' |
-    sed -e 's/^.*wiredtiger/wiredtiger/' |
+    sed -e 's/^.*wiredtiger/wiredtiger/' > $t_out_errors
+cat $t_out_errors |
     sed -e 's/:.*:.*://' |
-    sort > $t_out_filtered
-diff $t_diff_filtered $t_out_filtered > /dev/null && exit 0
+    LC_ALL=C sort > $t_out_filtered
 
+diff $t_diff_filtered $t_out_filtered > $t_diff_result && exit 0
 echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
 echo 'scan-build output differs from expected:'
 echo '<<<<<<<<<< Expected >>>>>>>>>> Current'
 echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
-diff $t_diff_filtered $t_out_filtered
+cat $t_diff_result
 echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
-echo 'scan-build output:'
+
+# Since we compare lines after stripping certain parts, it can be difficult to identify
+# the original source of the warnings. To recover the line numbers, we search for the
+# unexpected diff entries in the original output.
+echo 'that could be related to the following output entries:'
+echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+grep '^>' "$t_diff_result" | sed 's/^> *//' | while read line; do
+    path=${line%% *} # Cut off all at the first space
+    msg=${line#* } # Cut off all up to the first space
+    grep -F "$path:" "$t_out_errors" | grep -F "$msg"
+done
```

Comment:
(nitpick)
```suggestion
done | sort | uniq
```
This is to print only unique line+message. This helps in case if there are multiple instances of an error in a file.

Reply:
But then uniq wouldn't delete these entries as duplicates, would it?

And I am not sure about sort as well, since now we print the extended output in the same order as the errors from diff output which is better than alphabetical order from my point of view.

WDYT?